libsolidity/formal/CHC.cpp

```diff
@@ -1807,6 +1807,17 @@ tuple<CheckResult, smtutil::Expression, CHCSolverInterface::CexGraph> CHC::query
 	CheckResult result;
 	smtutil::Expression invariant(true);
 	CHCSolverInterface::CexGraph cex;
+	if (m_settings.printQuery)
+	{
+		if (auto smtLibInterface = dynamic_cast<CHCSmtLib2Interface*>(m_interface.get()))
```

Comment:
Same here regarding the check if `m_settings.solvers.smtlib2` is enabled and print a warning if it's not. Here there is an extra requirement, that z3 is off. So `smtlib2` should be the only solver enabled if the query is requested. In that case we can also assert the above.